keyboards/ergodox_ez/keymaps/davidhbrown/keymap.c

```diff
@@ -0,0 +1,99 @@
+
+  [2] = LAYOUT_ergodox(KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_CALCULATOR,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_NUMLOCK,KC_KP_7,KC_KP_8,KC_KP_9,KC_KP_0,KC_KP_MINUS,KC_TRANSPARENT,KC_TRANSPARENT,KC_KP_4,KC_KP_5,KC_KP_6,KC_KP_PLUS,KC_TRANSPARENT,KC_KP_SLASH,KC_KP_1,KC_KP_2,KC_KP_3,KC_KP_ASTERISK,KC_TRANSPARENT,KC_KP_ENTER,KC_TRANSPARENT,KC_KP_0,KC_KP_0,KC_KP_COMMA,KC_KP_DOT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_KP_SLASH,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT),
+
+  [3] = LAYOUT_ergodox(KC_AUDIO_MUTE,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_AUDIO_MUTE,KC_AUDIO_VOL_UP,KC_MS_BTN1,KC_MS_UP,KC_MS_BTN2,KC_MS_WH_UP,KC_MS_ACCEL2,KC_TRANSPARENT,KC_AUDIO_VOL_DOWN,KC_MS_LEFT,KC_MS_DOWN,KC_MS_RIGHT,KC_MS_WH_DOWN,KC_MS_ACCEL1,KC_TRANSPARENT,KC_MS_BTN3,KC_MS_BTN3,KC_MS_WH_LEFT,KC_MS_WH_RIGHT,KC_MS_ACCEL0,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_MS_WH_LEFT,KC_MS_WH_RIGHT,KC_MS_WH_UP,KC_TRANSPARENT,KC_TRANSPARENT,KC_MS_WH_DOWN,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_MS_WH_UP,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_MS_WH_LEFT,KC_MS_WH_DOWN,KC_MS_WH_RIGHT,KC_MS_ACCEL0,KC_MS_ACCEL1,KC_MS_ACCEL2,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_MS_BTN1,KC_MS_UP,KC_MS_BTN2,KC_TRANSPARENT,KC_MS_BTN3,KC_MS_LEFT,KC_MS_DOWN,KC_MS_RIGHT,KC_MEDIA_REWIND,KC_MEDIA_FAST_FORWARD,KC_MEDIA_NEXT_TRACK,KC_MEDIA_PREV_TRACK,KC_MEDIA_STOP,KC_MEDIA_PLAY_PAUSE),
+
+};
+
+rgblight_config_t rgblight_config;
+bool disable_layer_color = 0;
+
+bool suspended = false;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    switch (keycode) {
+      case EPRM:
```

Comment:
QMK has a keycode to do this built-in. You can use `EEP_RST` in place of `EPRM`.

https://github.com/qmk/qmk_firmware/blob/92f9b6c3bdff939bc562d640e02c3aebaa204e17/quantum/quantum.c#L323-L327
https://github.com/qmk/qmk_firmware/blob/92f9b6c3bdff939bc562d640e02c3aebaa204e17/quantum/quantum_keycodes.h#L613

Reply:
@noroadsleft this is from the default keymap, and needs to be updated (in fact, my ZSA PR updates this)

Reply:
I'm not sure what (if anything) I should be doing in response to these comments. Does it help that I don't use macro recording in the keymap? I think I could probably remove this function. (Yes, I did start with the default keymap.c plus the code arrays from the configurator).

Reply:
Yeah, it should be removed.